### PR TITLE
Add text extraction notebook and other demo2 files

### PR DESCRIPTION
**Pull request by Shreyanand**
For issue #11 and issue #26, add files required to run text extraction phase of demo 2. The extraction notebook is here: `notebooks/demo2/pdf_text_extraction.ipynb`. The source for extraction adapted from this [repository](https://github.com/os-climate/corporate_data_pipeline/tree/main/Allianz_NLP/esg_data_pipeline/esg_data_pipeline/components)  is in `src/components/preprocessing`.

Questions that remain to be addressed:

1. What would be the best way to give credit to the Allianz NLP repository?
2. The dataset currently is in the data folder of the repo, do we want to bring s3 in the picture at this point?

### CODE REVIEW

**Comment by erikerlandson** [comment]
@Shreyanand  lgtm

**Comment by MichaelClifford** [comment]
>Questions that remain to be addressed:

>What would be the best way to give credit to the Allianz NLP repository?
The dataset currently is in the data folder of the repo, do we want to bring s3 in the picture at this point?

@Shreyanand do these need to be addressed in this PR? or should they be turned into additional issues?

**Comment by MichaelClifford** [comment]
> The dataset currently is in the data folder of the repo, do we want to bring s3 in the picture at this point?

@Shreyanand, I am generally not a fan of using github as the place to store data (other than small example sets). This data is not huge, but I think it might be a good idea to put this data in remote storage and interactive with it that way. WDYT? 

@MichaelTiemannOSC is it all right to move this data and make it available through remote storage?

**Comment by MichaelTiemannOSC** [comment]
The Data Commons is designed around the concept of having access to cloud storage for just such a reason.  We should create a folder for this.  We just need to figure out an appropriate name for the folder that we can live with for a while.

**Comment by MichaelClifford** [comment]
@MichaelTiemannOSC great : ) 

Is this something that merits a new issue in the os_c_data_commons repo to discuss?

**Comment by erikerlandson** [comment]
@MichaelClifford  the following would be a good place to discuss:
https://github.com/os-climate/os_c_data_commons/issues/68

**Comment by MichaelClifford** [comment]
@MichaelTiemannOSC  According to the linked issue it seems like the ESG corporate data we are looking to use here is still a bit of an outlier. I'm happy with any naming convention/bucket location folks want to use, but would `corpdata/ESG/**`      in the 'ocp-odh-os-demo-s3' bucket suffice for now?

Let me know what you think :smile:

File: notebooks/demo2/pdf_text_extraction.ipynb [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import sys\n",
+    "\n",
+    "src_path = \"../../\"\n",
+    "if src_path not in sys.path:\n",
+    "    sys.path.insert(1, src_path)\n",
```
Comment (chauhankaranraj) [inline notebooks/demo2/pdf_text_extraction.ipynb:21]: If `src` is already added to Pipfile, do we still need to add it to path?

File: src/__init__.py [hunk 2]
```diff
@@ -1 +0,0 @@
-"""Analytics compiled into python code."""
```
Comment (chauhankaranraj) [inline src/__init__.py:1]: Would precommit complain if we remove this?

**Review by chauhankaranraj — CHANGES_REQUESTED** [review]
Just small nits otherwise /lgtm

**Comment by chauhankaranraj** [comment]
> @Shreyanand do these need to be addressed in this PR? or should they be turned into additional issues?

Agreed, let's discuss this in a separate issue - #39 - and unblock this PR :)

**Review by chauhankaranraj — APPROVED** [review]
changes look good, thanks :smile:

**Pull request merged**